pkg/nsx/services/subnet/store_test.go

```diff
@@ -0,0 +1,131 @@
+package subnet
+
+import (
+	"fmt"
+	"reflect"
+	"sync"
+	"testing"
+
+	"github.com/agiledragon/gomonkey"
+	"github.com/stretchr/testify/assert"
+	"github.com/vmware/vsphere-automation-sdk-go/runtime/bindings"
+	"github.com/vmware/vsphere-automation-sdk-go/runtime/data"
+	"github.com/vmware/vsphere-automation-sdk-go/services/nsxt/model"
+
+	"k8s.io/client-go/tools/cache"
+
+	"github.com/vmware-tanzu/nsx-operator/pkg/config"
+	"github.com/vmware-tanzu/nsx-operator/pkg/nsx"
+	"github.com/vmware-tanzu/nsx-operator/pkg/nsx/ratelimiter"
+	"github.com/vmware-tanzu/nsx-operator/pkg/nsx/services/common"
+)
+
+type fakeQueryClient struct {
+}
+
+func (qIface *fakeQueryClient) List(_ string, _ *string, _ *string, _ *int64, _ *bool, _ *string) (model.SearchResponse, error) {
+	cursor := "2"
+	resultCount := int64(2)
+	return model.SearchResponse{
+		Results: []*data.StructValue{&data.StructValue{}},
+		Cursor:  &cursor, ResultCount: &resultCount,
+	}, nil
+}
+
+func Test_IndexFunc(t *testing.T) {
+	id, tag, scope := "test_id", "test_tag", "nsx-op/subnet_cr_uid"
+	subnet := model.VpcSubnet{
+		Id:   &id,
+		Tags: []model.Tag{{Tag: &tag, Scope: &scope}},
+	}
+	t.Run("1", func(t *testing.T) {
+		got, _ := indexFunc(subnet)
```

Comment:
Don't forget to fix the UT failure before merging the patch: `vet: pkg/nsx/services/subnet/store_test.go:42:13: undeclared name: indexFunc`

Reply:
Thanks, done.